src/components/organisms/PostsContainerOrganism.tsx

```diff
@@ -0,0 +1,29 @@
+import { api } from "~/utils/api";
+import SyncLoader from "react-spinners/SyncLoader";
+
+const PostsContainerOrganism = () => {
+  const {data, isLoading} = api.posts.getAll.useQuery();
+  
+  // Early return
+  if(!data) {
+    return <div>Loading...</div>
+  }
+
+  if (isLoading) {
+    return <SyncLoader size={5} color={"#FFCD00"} />
+  }
+
+  return data.map(({post, author}) => (
+    <div key={post.id}className="p-8 border-b gap-4 border-slate-700 flex items-center">
+      <img 
+        className="w-12 h-12 rounded-full"
+        src={author?.imageUrl} 
+        alt="author avatar" 
+      />
+      <div className="">{post.content}</div>
+    </div>
+  ))
+
```

Comment:
```suggestion
return (
  <>
    { isLoading ? <SyncLoader size={5} color={"#FFCD00"} /> : posts }
  </>
)
```

上記の編集を行うことでこのような変更が加えることができるようになると思います！

型の問題もクリアできそうです！
<img width="1047" alt="image" src="https://github.com/maria-skygrid/t3-intro/assets/100871461/a3240059-79f8-4eb1-97de-e6d30c94a7e7">

Reply:
@qdi0 
なるほどです！今までreturn値をそこまで見てなかったです、、🙇‍♀️
各々コンポーネントをreturn値のtypeを一つにまとめないと、その後エラーが出てくる可能性があるのをなんとなく理解できました！これから気をつけます。理解しやすい説明ありがとうございます！🙇‍♀️